QRCoder/QRCodeGenerator/AlphanumericEncoder.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace QRCoder;
+
+public partial class QRCodeGenerator
+{
+    /// <summary>
+    /// Encodes alphanumeric characters (<c>0–9</c>, <c>A–Z</c> (uppercase), space, <c>$</c>, <c>%</c>, <c>*</c>, <c>+</c>, <c>-</c>, period, <c>/</c>, colon) into a binary format suitable for QR codes.
+    /// </summary>
+    private static class AlphanumericEncoder
+    {
+        private static readonly char[] _alphanumEncTable = { ' ', '$', '%', '*', '+', '-', '.', '/', ':' };
+
+        /// <summary>
+        /// A dictionary mapping alphanumeric characters to their respective positions used in QR code encoding.
+        /// This includes digits 0-9, uppercase letters A-Z, and some special characters.
+        /// </summary>
+        private static readonly Dictionary<char, int> _alphanumEncDict = CreateAlphanumEncDict(_alphanumEncTable);
+
+        /// <summary>
+        /// Creates a dictionary mapping alphanumeric characters to their respective positions used in QR code encoding.
+        /// This includes digits 0-9, uppercase letters A-Z, and some special characters.
+        /// </summary>
+        /// <returns>A dictionary mapping each supported alphanumeric character to its corresponding value.</returns>
+        private static Dictionary<char, int> CreateAlphanumEncDict(char[] alphanumEncTable)
+        {
+            var localAlphanumEncDict = new Dictionary<char, int>(45);
+            // Add 0-9
+            for (char c = '0'; c <= '9'; c++)
+                localAlphanumEncDict.Add(c, c - '0');
```

Comment:
Changed from `localAlphanumEncDict.Add($"{i}"[0], i);` to eliminate string encoding